tests/run-make/crate-loading/multiple-dep-versions.stderr

```diff
@@ -92,6 +93,7 @@ LL |     do_something(OtherType);
    |     |
    |     required by a bound introduced by this call
    |
+   = note: `OtherType` implements similarly named `dependency::Trait`, but not `Trait`
```

Comment:
"there are multiple different versions of crate `dependency` in the dependency graph" is a simply *more specific* message than "`OtherType` implements similarly named `dependency::Trait`, but not `Trait`", is it not?

I personally really don#t think we should be emitting the new msg if we already detected that there are two different versions of the same crate